agent/engine/docker_task_engine_test.go

```diff
@@ -1035,6 +1035,15 @@ func TestProvisionContainerResourcesSetPausePIDInVolumeResources(t *testing.T) {
 		}, nil),
 		mockCNIClient.EXPECT().SetupNS(gomock.Any(), gomock.Any(), gomock.Any()).Return(nsResult, nil),
 	)
+	// These mock calls would be made only for Windows.
```

Comment:
hmm.. this is Hacky, the test will succeed in Windows even if you remove the calls to these, right?

Reply:
Thinking about this, it probably would be better to extract `invokeCommandsForTaskNamespaceSetup` and `invokeCommandsForTaskNamespaceCleanup` outside of `DockerTaskEngine`. Perhaps we can create an interface like 
```golang
type TaskNamespaceHelper struct {
    NamespaseSetup() error
    NamespaceCleanup() error
}
```

This way the UTs will only validate that the high-level `NamespaseSetup` and `NamespaceCleanup` are called and the windows implementation UT can check the docker operations are called exactly once, whereas the Linux implementation will be empty.

Thoughts?

Reply:
I agree with you. I have moved these methods to a separate interface in `ecscni` package.